src/presign/round_three.rs

```diff
@@ -20,25 +20,88 @@ use crate::{
         Proof, ProofContext,
     },
 };
-use k256::{elliptic_curve::PrimeField, Scalar};
+use k256::{
+    elliptic_curve::{subtle::CtOption, PrimeField},
+    Scalar,
+};
 use libpaillier::unknown_order::BigNumber;
 use merlin::Transcript;
 use serde::{Deserialize, Serialize};
-use std::fmt::Debug;
+use std::{
+    fmt::Debug,
+    ops::{AddAssign, Deref},
+};
 use tracing::error;
 use zeroize::ZeroizeOnDrop;
+//use libpaillier::unknown_order::CtOption;
 
-#[derive(Clone, ZeroizeOnDrop)]
+#[derive(Clone)]
 pub(crate) struct Private {
-    pub k: BigNumber,
-    pub chi: Scalar,
-    #[zeroize(skip)]
+    pub k: SecretBigNumber,
+    pub chi: SecretScalar,
     pub Gamma: CurvePoint,
-    #[zeroize(skip)]
-    pub delta: Scalar,
-    #[zeroize(skip)]
+    pub delta: SecretScalar,
     pub Delta: CurvePoint,
 }
+#[derive(Clone, ZeroizeOnDrop)]
+pub(crate) struct SecretBigNumber(BigNumber);
+
+impl SecretBigNumber {
+    pub fn get_bignumber(&self) -> &BigNumber {
```

Comment:
Similar comment as my prev comment about changing the name and adding documentation warning against cloning.